### PR TITLE
Support for updating task definitions

### PR DESCRIPTION
**Pull request by fishnix**
* Add the endpoint for updating task definitions.  
* Split out sharable code for processing repository credentials and start to refactor it into smaller pieces where possible.  It's still a mess and way more complex than I would like.
* Fixed an issue where the new `spinup:category` tag was being applied to space level dependent resources (clusters, roles)

### CODE REVIEW

**Comment by btassone** [comment]
Haven't checked the update yet but still missing these 2 minor things:

* Missing readme for update task def route
* DELETE and GET routes are incorrect - cluster should be clusters

**Comment by btassone** [comment]
Tested out the change. Now works as intended!

File: api/handlers_taskdef.go [hunk 1]
```diff
@@ -175,3 +175,52 @@ func (s *server) TaskDefShowHandler(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	w.Write(j)
 }
+
+func (s *server) TaskDefUpdateHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (damnski) [inline api/handlers_taskdef.go:179]: :eyes: - missing func description

File: orchestration/orchestration_taskdef.go [hunk 2]
```diff
@@ -103,6 +121,65 @@ func (o *Orchestrator) CreateTaskDef(ctx context.Context, input *TaskDefCreateOr
 	return output, nil
 }
```
Comment (damnski) [inline orchestration/orchestration_taskdef.go:123]: :eyes:

**Review by damnski — COMMENTED** [review]
looks good, just a couple comments on missing comments.

**Review by FishNecks — APPROVED** [review]
👍

**Pull request merged**